src/Chapter3.hs

```diff
@@ -374,6 +389,46 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Fighter = MkFighter
+    { fighterHealthPts    :: Int,
+      fighterAttackPts    :: Int,
+      fighterGold         :: Int
+    } deriving (Show)
+
+knight :: Fighter
+knight = MkFighter
+    { fighterHealthPts = 10,
+      fighterAttackPts = 2,
+      fighterGold = 5
+    }
+
+monster :: Fighter
+monster = MkFighter
+    { fighterHealthPts = 15,
+      fighterAttackPts = 1,
+      fighterGold = 11
+    }
+
+
+fight :: Fighter -> Fighter -> Int
+fight kn mn 
+    | fighterHealthPts kn < 1 = if fighterHealthPts mn < 1 then fighterGold kn else -1
+    | fighterHealthPts mn < 1 = (fighterGold kn) + (fighterGold mn)
+    | fighterAttackPts kn < 1 = if fighterAttackPts mn < 1 then fighterGold kn else -1
+    | fighterAttackPts mn < 1 = (fighterGold kn) + (fighterGold mn)
+    | otherwise = if knToDeathRounds kn mn >= mnToDeathRounds kn mn then (fighterGold kn) + (fighterGold mn) else -1
+    where 
+        knToDeathRounds :: Fighter -> Fighter -> Int
+        knToDeathRounds kn mn = if (fighterHealthPts kn) `rem` (fighterAttackPts mn) > 0 then (fighterHealthPts kn) `div` (fighterAttackPts mn) + 1 else (fighterHealthPts kn) `div` (fighterAttackPts mn)
```

Comment:
I think this function is a bit overcomplicated 🙂 
What you actually need to check is that the health of one fighter is bigger than the attack of another fighter. So you can make two checks to see if that the case and act accordingly 👌🏼